torch/testing/_internal/common_utils.py

```diff
@@ -1732,6 +1762,19 @@ def skip_exception_type(exc_type):
     except exc_type as e:
         raise unittest.SkipTest(f"not implemented: {e}") from e
 
+@contextmanager
+def print_repro_on_failure(repro_str):
+    try:
+        yield
+    except unittest.SkipTest:
+        raise
+    except Exception as e:
+        # NB: Hacking the exception args is the cleanest way I've found to append
+        # failure reproduction info without poisoning the stack trace.
```

Comment:
I suppose the other possibility is to modify the overall test runner. This is how I added the hint for 'keep-going':

```
        if (
            not options.continue_through_error
            and not RERUN_DISABLED_TESTS
            and len(failure_messages) != 0
        ):  
            raise RuntimeError(
                "\n".join(failure_messages)
                + "\n\nTip: You can keep running tests even on failure by "
                "passing --keep-going to run_test.py.\n"
                "If running on CI, add the 'keep-going' label to "
                "your PR and rerun your jobs."
            )
```

This is done in run_test.py, so it will get printed even if the testing process crashes.

Reply:
I'd like to be consistent with that, but I don't think I have access to the right stuff at the `run_test.py` level. AFAICT the test runner isn't aware of individual test case names.

Also it's kinda nice to get the repro output when running outside of a `run_test.py` context as well.